staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -309,6 +319,32 @@ func (w *watchCache) updateCache(event *watchCacheEvent) {
 	w.endIndex++
 }
 
+// Use dynamic size cache. If all cached events occur inside eventFreshDuration, increase cache capacity by 2x.
+// Decrease cache capacity by 2x when recent half events occur outside eventFreshDuration.
+func (w *watchCache) resizeCache(event *watchCacheEvent) {
+	if w.endIndex == w.startIndex+w.capacity && event.RecordTime.Sub(w.cache[w.startIndex%w.capacity].RecordTime) < eventFreshDuration {
+		if w.capacity*2 < maxEventCapacity {
```

Comment:
If we set base value to 100, this effectively means that our max capacity will be at 102400, not at 150000.

Here, you should take min(2*capacity, maxEventCapacity)

Reply:
addressed